pytorch3dunet/datasets/hdf5.py

```diff
@@ -84,14 +86,14 @@ def __getitem__(self, idx):
         if idx >= len(self):
             raise StopIteration
 
-        # get the slice for a given index 'idx'
         raw_idx = self.raw_slices[idx]
-        # get the raw data patch for a given slice
-        raw_patch_transformed = self.raw_transform(self.raw[raw_idx])
+        raw_idx_padded = tuple(slice(this_index.start, this_index.stop + 2 * this_halo, None) for this_index, this_halo in zip(raw_idx, self.halo_shape))
```

Comment:
I belie it should be: `this_index.start - this_halo, this_index.stop + this_halo` inside the loop.

You're approach should still work, but we should build slices based on the `self.raw_padded` volume instead of the `this.raw`, see lines:
```
# build slice indices for raw and label data sets
slice_builder = get_slice_builder(self.raw, self.label, self.weight_map, slice_builder_config)
```

Reply:
`raw_idx` is for `raw` with its original shape, `raw_idx_padded` is used on `raw_padded` where all sides are padded but its index still starts from 0. `this_index.start - this_halo, this_index.stop + this_halo` will start the slice from negative number. My current implementation is correct I think.

Reply:
I added a test to check if the halo-padded input generated from `StandardHDF5Dataset` equals to the original slices of raw after removal of halo.